django/__init__.py

```diff
@@ -21,4 +30,16 @@ def setup(set_prefix=True):
         set_script_prefix(
             "/" if settings.FORCE_SCRIPT_NAME is None else settings.FORCE_SCRIPT_NAME
         )
-    apps.populate(settings.INSTALLED_APPS)
+
+    with ExitStack() as stack:
+        for connection in connections.all():
+            stack.enter_context(connection.execute_wrapper(_warn_on_queries))
+        apps.populate(settings.INSTALLED_APPS)
+
+
+def _warn_on_queries(execute, *args):
+    warnings.warn(
+        "Accessing the database during app initialization is discouraged.",
```

Comment:
I believe that this message should offer an alternative to users or a link to docs with an alternative to properly fix the warning. See my next comment :-).